File: frontend/src/types/notification.ts
export interface NotificationItem {
  id: string
  user_id: string
  type: string
  title: string
  body: string | null
  action_url: string | null
  related_media_id: string | null
  related_user_id: string | null
  is_read: boolean
  read_at: string | null
  sent_at: string | null
  created_at: string
}

export interface NotificationListResponse {
  items: NotificationItem[]
  total: number
  limit: number
  offset: number
}

export interface NotificationMarkReadResponse {
  updated_count: number
}

export interface NotificationPreferences {
  user_id: string
  new_episode: boolean
  new_chapter: boolean
  friend_activity: boolean
  recommendations: boolean
  watch_party_invite: boolean
  watch_party_reminder: boolean
  discord_webhook: string | null
  telegram_chat_id: string | null
  email_enabled: boolean
  push_enabled: boolean
  updated_at: string
}

export interface NotificationPreferencesUpdateRequest {
  new_episode?: boolean
  new_chapter?: boolean
  friend_activity?: boolean
  recommendations?: boolean
  watch_party_invite?: boolean
  watch_party_reminder?: boolean
  discord_webhook?: string | null
  telegram_chat_id?: string | null
  email_enabled?: boolean
  push_enabled?: boolean
}

File: backend/src/app/schemas/notification.py
"""Schemas for notification endpoints (Phase 11)."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from src.app.models.enums import NotificationType


class NotificationResponse(BaseModel):
    """Single notification item in user inbox."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    type: NotificationType
    title: str
    body: str | None = None
    action_url: str | None = None
    related_media_id: UUID | None = None
    related_user_id: UUID | None = None
    is_read: bool
    read_at: datetime | None = None
    sent_at: datetime | None = None
    created_at: datetime


class NotificationListResponse(BaseModel):
    """Paginated notifications visible to current user."""

    items: list[NotificationResponse]
    total: int
    limit: int
    offset: int


class NotificationMarkReadRequest(BaseModel):
    """Payload for marking notification items as read."""

    notification_ids: list[UUID] = Field(default_factory=list)


class NotificationMarkReadResponse(BaseModel):
    """Response for mark-as-read operation."""

    updated_count: int
